pkg/buildah/login.go

```diff
@@ -59,14 +68,58 @@ func newLoginCommand() *cobra.Command {
 		Use:   "login",
 		Short: "Login to a container registry",
 		Long:  loginDescription,
+
+		PreRunE: func(cmd *cobra.Command, args []string) (err error) {
+			if len(args) == 0 {
+				args = append(args, DefaultRegistry)
+			}
+
+			if args[0] == DefaultRegistry {
```

Comment:
We can support any host wether they support `sealos` login or not.
Just parse user inputs, and do login, and save response.

If user want to login to `docker.io`, they can just use command: `sealos login docker.io -u xxx -p xxx` as usual, alone with user can login into `hub.sealos.io` witch currently support `sealos login hub.sealos.io -u xxx -k path/to/file`.

With this design, we can both support old behavior and plus support `kubeconfig` login. Further more, once our `hub.sealos.io` updated support  username+password login(which we can do but not do right now), users can login any registries as they need using the same method.

Reply:
> sealos login hub.sealos.io -u xxx -k path/to/file.
here, we can parse kubeconfig and get user.